roles/metrics-export.lua

```diff
@@ -258,25 +298,47 @@ local function apply_http(conf)
     local enabled = {}
     for _, node in ipairs(conf) do
         if #(node.endpoints or {}) > 0 then
-            local host, port, err = parse_listen(node.listen)
-            if err ~= nil then
-                error("failed to parse URI: " .. err, 2)
+            local host, port, target
+            if node.server ~= nil then
+                local server = httpd_role.get_server(node.server)
+                host, port = server.host, server.port
+                target = node.server
+            elseif node.listen ~= nil then
+                target = node.listen
+                local err
+                host, port, err = parse_listen(node.listen)
+                if err ~= nil then
+                    error("failed to parse URI: " .. err, 2)
+                end
+            else
+                local server = httpd_role.get_server()
+                host, port = server.host, server.port
+                target = httpd_role.DEFAULT_SERVER_NAME
             end
-            local listen = node.listen
 
             http_servers = http_servers or {}
-            enabled[listen] = true
+            enabled[target] = true
```

Comment:
In theory, `listen` and `target` values may clash. So we need some logic to avoid it.

Reply:
Fixed this problem by using concatenation `target` and `is_httpd_role` values as a key to have a unique values.